dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/OracleMetadataConnector.java

```diff
@@ -45,43 +45,17 @@ public OracleMetadataConnector() {
     super(OracleConnectorScope.METADATA);
   }
 
-  static interface GroupTask<T> extends Task<T> {
-
-    @CheckForNull
-    public Exception getException();
-  }
-
-  static class SelectTask extends JdbcSelectTask implements GroupTask<Summary> {
-
-    private Exception throwable;
-
-    public SelectTask(@Nonnull String file, @Nonnull String selectQuery) {
-      super(file, selectQuery);
-    }
-
-    @Override
-    public boolean handleException(Exception e) {
-      throwable = e;
-      return true;
-    }
-
-    @Override
-    public Exception getException() {
-      return throwable;
-    }
-  }
-
   private static void addAtLeastOneOf(
-      @Nonnull List<? super Task<?>> out, @Nonnull GroupTask<?>... tasks) {
-    for (GroupTask<?> task : tasks) out.add(Preconditions.checkNotNull(task));
+      @Nonnull List<? super Task<?>> out, @Nonnull GroupTask... tasks) {
+    for (GroupTask task : tasks) out.add(Preconditions.checkNotNull(task));
     Task<?> msg_task = MessageTask.create(tasks).onlyIfAllFailed(tasks);
     out.add(msg_task);
   }
 
   @Nonnull
-  private static SelectTask newSelectStarTask(
+  private static GroupTask newSelectStarTask(
       @Nonnull String file, @Nonnull String table, @Nonnull String where) {
-    return new SelectTask(file, "SELECT * FROM " + table + where);
+    return GroupTask.createSelect(file, "SELECT * FROM " + table + where);
```

Comment:
I would suggest adding a space between `table` and `where` instead of requiring that one of these variables starts/ends with the space.

If it is desirable for multiple consecutive spaces to be squashed, then I would recomment the method similar to `formatQuery`.
https://github.com/google/dwh-migration-tools/blob/eaeadef2449b465b59633f2034ea55104e70bf01/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataUtils.java#L44

Reply:
Acknowledged - this is a good point, but I prefer to avoid touching the logic that builds oracle-metadata SQLs in this PR, because it seems to risky. I'll add a Preconditions check to make sure that the end of `table` and start of `where` don't get glued into one word.

Adding an extra space would break the tests, because a string comparison would start failing.

`formatQuery` would work to fix the query, but what I'd like to do here is ensure that the query is already correct when it's built.